File: client/schema/code_review.ts
export interface CodeReviewFile {
    id: string;
    project_id: string;
    original_name: string;
    stored_name: string;
    file_path: string;
    created_at: Date;
}

export interface CodeReviewFileUpdate {
    original_name: string;
}

export interface CodeReviewGenerate {
    project_id: string;
    user_prompt: string;
}

File: api/schemas/code_review.py
from pydantic import BaseModel, UUID4
from datetime import datetime
from typing import Optional

class CodeReviewBase(BaseModel):
    id: UUID4
    project_id: UUID4
    original_name: str
    stored_name: str
    file_path: str
    created_at: datetime

    class Config:
        from_attributes = True

class CodeReviewCreate(BaseModel):
    project_id: UUID4
    original_name: str
    stored_name: str
    file_path: str

    class Config:
        from_attributes = True

class CodeReviewUpdate(BaseModel):
    original_name: Optional[str] = None

    class Config:
        from_attributes = True

class CodeReviewGenerate(BaseModel):
    project_id: UUID4
    user_prompt: str